src/mappings/gns.ts

```diff
@@ -191,6 +193,18 @@ export function handleSubgraphPublished(event: SubgraphPublished): void {
   subgraphVersion.metadataHash = event.params.versionMetadata
   subgraphVersion = fetchSubgraphVersionMetadata(subgraphVersion, base58Hash)
   subgraphVersion.save()
+
+  let oldDeployment: SubgraphDeployment | null = null
+  if (oldVersionID != null) {
+    let oldVersion = SubgraphVersion.load(oldVersionID)
+    oldDeployment = SubgraphDeployment.load(oldVersion.subgraphDeployment)
+  }
+  // create deployment - named subgraph relationship, and update the old one
+  updateCurrentDeploymentLinks(
```

Comment:
I think this function, or one similar to it, will have to run on `handleSubgraphDeprecated`. it might be that we just need to lower the active count, and set active to false. wdyt?

Reply:
🤔 
maybe, yeah, although technically the subgraph still has the deployment as its' current version, so it might be weird when querying that data 🤔

Reply:
Maybe it would be best to track that as a separate count, like having `subgraphCount`, `activeSubgraphCount` and `deprecatedSubgraphCount`, since an active subgraph for the deployment can stop being active because it was upgraded or because it was deprecated